language-support/java/codegen/src/main/scala/com/digitalasset/daml/lf/codegen/backend/java/inner/FromJsonGenerator.scala

```diff
@@ -63,7 +63,7 @@ private[inner] object FromJsonGenerator extends StrictLogging {
     val typeName = className.parameterized(typeParams)
 
     val argNames = {
-      val names = fields.map(f => CodeBlock.of("$S", f.javaName))
```

Comment:
This was a bug, uncovered by a key type (`PartyAndInt`) that had a field called `int`. In the json form, this should appear as an object field `"int"`, but we were using the java-sanitized version `"int$"`, which json doesn't need to know about.